### DISCUSSION TITLE
Document backup strategies and restore requirements

### DISCUSSION THREAD

**Discussion by goebbe** [borgbase/vorta#821] [Ideas]
**Is your feature request related to a problem? Please describe.**
Depending on how Vorta/ borgbackup is used, there are different **requirements for backup recovery**.
To my knowledge, there is no documentation of the requirements and there is a risk that in case one needs to recover, the user is misses some or all required information and is consequently not able to recover. 
Power-users are probably aware of this - but average users that a new to Vorta and remote backup are likely to find this useful. 

**Describe the solution you'd like**
Document the requirements for different scenarios.
Make it easy to export/print all required information.

**Describe alternatives you've considered**
Leave this undocumented i.e. learning by experience in case you need to recover.

**Comment by goebbe** [comment]
The following text may be useful to get this started: 

--------------------------------------------------------------------------------------
Backup strategies for newbies - and requirements to restore data: 
---------------------------------------------------------------------------------------
Backups should help to prevent data loss. However, depending on the scenario, there are different backup-strategies, each of them leads to different requirements to restore your data. 

Scenario A. Files on your computer are deleted/lost
---------------------------------------------------

Minimum backup strategy: 
- backup (copy of the files) on your computer (e.g. the same disk)

Condition precedent to restore the data
- path and name of the repository (on your computer)
- passphrase of the repository in order to restore the backup

If you use Vorta, nothing special is required. You could mount your backup-repository using Vorta and browse/copy/paste your files from the last backup.


Scenario B. Disk failure/ virus on your main computer, unable to access Vorta and the files on the disk
-------------------------------------------------------------------------------------------------------

Minimum backup strategy: 
- backup on an (separate) backup-disk (e.g. usb-disk that is used for your backup)
- this also covers scenario A

Condition precedent to restore the data: 
1 access to a computer with Vorta/ borgbackup and installed
2 connect the backup-disk to this computer
3 path and name of the (backup) repository
4 passphrase of the repository in order to restore the backup

Information about 3 and 4 should be accessible, even if your disk is broken. 
Restore requires storage of the necessary information at a save place, outside your computer/main disk!


Scenario C. Computer equipment is destroyed or stolen (e.g. your house burns down)
---------------------------------------------------------------------------------

Minimum backup strategy: 
- remote backup e.g. on a remote server
- this also covers scenario A and B

Condition precedent to restore the data: 
1 access to a computer with Vorta/ borgbackup and ssh installed
2 address of the server (URL), login and password of the server (ssh) in order to access the remote server
3 path and name of the repository (on the server)
4 passphrase of the repository in order to restore the backup

Information about 2, 3 and 4 should be accessible, even if your electronic equipment is destroyed. 
This scenario requires you to keep the information at a save place, outside your house!

--------------------------------------
Note: If you use Vorta on Linux, the passphrase of the repository is stored in your keyring of your OS (operating system) 

Recommendation: 
------------------------
- Save all the necessary information using a **password-manager**.
- Keep a **copy of your (encrypted) passwords** at a save place, possibly **outside** your home - and make sure to use a save password for the password-manager, that you can remember. :-)
- **Test** that you are actually able to recover - and document necessary steps.

**Comment by m3nu** [comment]
All correct. [Vorta's docs](https://vorta.borgbase.com) are in[ another repo](https://github.com/borgbase/vorta.borgbase.com). Question is if this information fits there or if it's better kept with Borg, since it's more high level.

We already mention similar things on the [Restore page](https://github.com/borgbase/vorta.borgbase.com/blob/master/usage/restore.md).

**Reply by quazgar** [reply]
Would you be open to a PR for an additional section in the documentation for first steps to restore from the commandline? For when Vorta's configuration is also destroyed and maybe no Vorta instance is available.

**Reply by goebbe** [reply]
@quazgar  I think it is reasonable to restrict the docs to recovery using Vorta, i.e. a computer running Vorta is one of the requirements to restore  (as described in the docs).  Perhaps one could point to the possibility to use the command line only  (with borg and ssh) - but this requires deeper knowledge and is already described in the docs of borg.

**Comment by goebbe** [comment]
Of course most of the text does apply to borgbackup - some parts, however, are specific to Vorta: 
- Scenario A:  Using **Vorta**, there are no special requirements  - just mount the last backup and recover your files!!  :-) 
- Using ** Vorta**, the repo passprhase is saved in the keyring of the OS! This is not easy to discover (at least it took me some time to understand where I can find my passphrase)
- The users-base of Vorta is probably different from the user-base of borgbackup-scripts. 

**Suggestion**: Add an **option** to Vorta, that exports/print all information that is necessary to recover to a text file.
- path or address of the server (URL), login and password/ keyfile of the server (ssh) in order to access the remote server
- path and name of the repository (on the server)
- passphrase of the repository (from keyring) in order to restore the backup

**Reply by m3nu** [reply]
We already have a feature to export all those settings to a JSON file. It also has an option to include the password IIRC. Could this serve the same purpose?

**Reply by goebbe** [reply]
@m3nu Thanks for the pointer. I will check/ test the export of settings for restore - and report, here. Does the export of settings cover all cases? e.g. remote backup via ssh? Anyway, this is a good opportunity to test if I would be able to restore from a remote backup (from a different computer). @quazgar Perhaps you want to joint the test - and document missing information (if there is information missing).

**Reply by goebbe** [reply]
I opened a new "discussion", where I document/ share my restore experience:  https://github.com/borgbase/vorta/discussions/1921
